apps/central-scan/frontend/src/app_root.tsx

```diff
@@ -418,97 +321,69 @@ export function AppRoot({
-        } Ballot Mode...`,
-      });
-      await setTestModeMutation.mutateAsync({ testMode: !isTestMode });
-      await logger.log(LogEventId.ToggledTestMode, userRole, {
-        disposition: 'success',
-        message: `Successfully toggled to ${
-          isTestMode ? 'Official' : 'Test'
-        } Ballot Mode`,
-      });
-      history.replace('/');
-    } catch (error) {
-      assert(error instanceof Error);
-      await logger.log(LogEventId.ToggledTestMode, userRole, {
-        disposition: 'failure',
-        message: `Error toggling to ${
-          isTestMode ? 'Official' : 'Test'
-        } Ballot Mode: ${error.message}`,
-        result: 'Mode not changed, user shown error.',
-      });
-    } finally {
-      setTogglingTestMode(false);
-    }
-  }, [logger, userRole, isTestMode, setTestModeMutation, history]);
-
-  const setMarkThresholdOverrides = useCallback(
-    async (markThresholdOverrides?: MarkThresholds) => {
-      await config.setMarkThresholdOverrides(markThresholdOverrides);
-      await refreshConfig();
-      history.replace('/');
-    },
-    [history, refreshConfig]
-  );
-
+  // poll for scanning status on an interval if configured
   useInterval(
     useCallback(async () => {
-      if (electionDefinition) {
+      if (electionDefinitionQuery.data) {
         await updateStatus();
       }
-    }, [electionDefinition, updateStatus]),
+    }, [electionDefinitionQuery.data, updateStatus]),
     1000
   );
 
+  // initial scanning status check
   useEffect(() => {
     void updateStatus();
   }, [updateStatus]);
 
+  // close the configure success screen if the USB drive is ejected
+  const wasJustConfigured = !!configureMutation.data;
   useEffect(() => {
-    if (electionJustLoaded && usbDrive.status === 'ejected') {
-      setElectionJustLoaded(false);
+    if (
+      wasJustConfigured &&
+      !configureSuccessScreenClosed &&
+      usbDrive.status === 'ejected'
+    ) {
+      setConfigureSuccessScreenClosed(false);
     }
-  }, [electionJustLoaded, usbDrive.status]);
+  }, [configureSuccessScreenClosed, usbDrive.status, wasJustConfigured]);
 
+  // auto-configure from USB drive
   useEffect(() => {
-    async function configure() {
-      if (
-        !configureMutation.isLoading &&
-        !electionDefinition &&
-        authStatusQuery.data &&
-        isElectionManagerAuth(authStatusQuery.data) &&
-        usbDrive.status === 'mounted'
-      ) {
-        const result = await configureMutation.mutateAsync();
-        const electionDefinitionFromResult = result.ok();
-        assert(electionDefinitionFromResult !== undefined);
-        updateElectionDefinition(electionDefinitionFromResult);
-      }
+    if (
+      electionDefinitionQuery.isSuccess &&
+      !electionDefinitionQuery.data &&
+      authStatusQuery.data &&
+      isElectionManagerAuth(authStatusQuery.data) &&
+      usbDrive.status === 'mounted' &&
+      !configureMutation.isLoading
+    ) {
+      configureMutation.mutate();
```

Comment:
I simplify the logic here, but it still has an infinite retry sort of bug identified in m17a QA which led to #3633. I address this in the final commit after manual testing.